packages/firestore/src/local/local_store.ts

```diff
@@ -183,7 +183,11 @@ export class LocalStore {
   start(): Promise<void> {
     // TODO(multitab): Ensure that we in fact don't need the primary lease.
     return this.persistence.runTransaction('Start LocalStore', false, txn => {
-      return this.startMutationQueue(txn).next(() => this.startQueryCache(txn));
+      return PersistencePromise.waitFor([
+        this.startMutationQueue(txn),
+        this.startQueryCache(txn),
+        this.startRemoteDocumentCache(txn)
```

Comment:
Hrm...  I'm nervous about doing these in parallel.  Generally we do one thing at a time to keep our "threading" model simple and not have to worry about concurrent execution of things.  Maybe this is safe to parallelize but if you're not 100% sure or there isn't a meaningful perf reason to do it in parallel, I suggest putting it back. :-)

Reply:
Done. I just mostly didn't like the way this looked, but cosmetics shouldn't be the main factor here.